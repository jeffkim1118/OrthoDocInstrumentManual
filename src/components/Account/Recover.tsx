import { Form, Field } from "react-final-form";
import { FORM_ERROR } from "final-form";
import { Button } from "react-bootstrap";
import { Link } from "react-router-dom";

export default function Recover() {
  const onSubmit = async (e: any) => {
    e.preventDefault();
  };

  return (
    <div className="background">
      <div className="recover-form-background">
      <div
        className="recover-form-container"
        
      >
        <Form
          onSubmit={onSubmit}
          validate={(values) => {
            const errors: any = {};
            if (!values.email) {
              errors.email = "Email required";
            }
            return errors;
          }}
          render={({
            submitError,
            handleSubmit,
            form,
            submitting,
            pristine,
            values,
          }) => (
            <form className="recover-password-form">
              <h3 className="recover-label">Forgot Your Password?</h3>
              <p className="recover-label">Don't fret! Just type in your email and we will <br/><span>send you a code to reset your password!</span></p>
              <Field name="email">
                {({ input, meta }) => (
                  <div className="recover-form">
                    <input
                      {...input}
                      type="email"
                      placeholder="Enter your email"
                      className={meta.touched && meta.error ? "error" : "recover-input"}
                    ></input>
                    <br />
                    {(meta.error || meta.submitError) && meta.touched && (
                      <span className="error-msg">
                        {meta.error || meta.submitError}
                      </span>
                    )}
                  </div>
                )}
              </Field>

              <Button
                className="recover-btn"
                type="submit"
                disabled={submitting || pristine}
              >
                Submit
              </Button>
              <div style={{ textAlign: "center" }}>
                <Link to={"/login"}>Go back</Link><br/>
                <Link to={"/signup"}>Don't have an account?</Link>
              </div>
            </form>
          )}
        ></Form>
      </div>
      </div>
    </div>
  );
}
