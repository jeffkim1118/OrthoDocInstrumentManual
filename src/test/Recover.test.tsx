import { BrowserRouter } from "react-router-dom";
import Recover from "../components/Account/Recover";
import { render, fireEvent, screen} from "@testing-library/react";

const renderRecover = () => {
    const recover = render(<BrowserRouter><Recover/></BrowserRouter>);
    return recover
}

describe("The recover component " , ()=>{
    it(" should render", () => {
        const {container} = renderRecover();
        expect(container).toMatchSnapshot();
    })

    it('should display labels', ()=> {
        renderRecover();
        const label1 = screen.getByText('Forgot Your Password?');
        const label2 = screen.getByText("Don't fret! Just type in your email and we will");
        expect(label1).toBeInTheDocument();
        expect(label2).toBeInTheDocument();
    })

    it(" should render input", () => {
        renderRecover();
        const recoverPasswordInput = screen.getByPlaceholderText('Enter your email');
        expect(recoverPasswordInput).toBeInTheDocument();
    })
    it("the submit button should be disabled when there's nothing in the input", () => {
        renderRecover();
        const registerBtn = screen.getByText("Submit");
        const emailPlaceholder = screen.getByPlaceholderText('Enter your email');
        fireEvent.change(emailPlaceholder, {target: {value: ""}});
        expect(registerBtn).toBeDisabled();
    })
    it("error message should appear when the input was clicked but nothing was entered after unfocusing", () => {
        renderRecover();
        const emailPlaceholder = screen.getByPlaceholderText('Enter your email');
        fireEvent.change(emailPlaceholder, {target: {value: ""}});
        fireEvent.focusIn(emailPlaceholder);
        fireEvent.focusOut(emailPlaceholder);
        const errorMsg = screen.getByText("Email required");
        expect(errorMsg).toBeInTheDocument()
    })
})