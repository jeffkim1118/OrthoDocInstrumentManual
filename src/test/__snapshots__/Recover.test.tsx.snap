// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The recover component   should render 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="recover-form-background"
    >
      <div
        class="recover-form-container"
      >
        <form
          class="recover-password-form"
        >
          <h3
            class="recover-label"
          >
            Forgot Your Password?
          </h3>
          <p
            class="recover-label"
          >
            Don't fret! Just type in your email and we will 
            <br />
            <span>
              send you a code to reset your password!
            </span>
          </p>
          <div
            class="recover-form"
          >
            <input
              class="recover-input"
              name="email"
              placeholder="Enter your email"
              type="email"
              value=""
            />
            <br />
          </div>
          <button
            class="recover-btn btn btn-primary"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <div
            style="text-align: center;"
          >
            <a
              href="/login"
            >
              Go back
            </a>
            <br />
            <a
              href="/signup"
            >
              Don't have an account?
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
